compiler/rustc_traits/src/chalk/lowering.rs

```diff
@@ -480,6 +482,7 @@ impl<'tcx> LowerInto<'tcx, Ty<'tcx>> for &chalk_ir::Ty<RustInterner<'tcx>> {
                     substs: application_ty.substitution.lower_into(interner),
                     item_def_id: assoc_ty.0,
                 }),
+                chalk_ir::TypeName::Foreign(def_id) => ty::Foreign(def_id.0),
```

Comment:
Will want to add the other way around too.

Reply:
Could you elaborate?

Reply:
We want to lower *to* Chalk here: https://github.com/rust-lang/rust/blob/fb9de9a8f082cb6bb046f3cd0a9e52768ed378ff/compiler/rustc_traits/src/chalk/lowering.rs#L273

Reply:
I see, I gave that a try.